subgraphs/uniswap-forks/src/common/creators.ts

```diff
@@ -41,28 +41,46 @@ import { getTrackedVolumeUSD } from "../price/price";
  * Create the fee for a pool depending on the the protocol and network specific fee structure.
  * Specified in the typescript configuration file.
  */
-export function createPoolFees(poolAddress: string): string[] {
-  const poolLpFee = new LiquidityPoolFee(poolAddress.concat("-lp-fee"));
-  const poolProtocolFee = new LiquidityPoolFee(
+export function createPoolFees(
+  poolAddress: string,
+  blockNumber: BigInt
+): string[] {
+  // get or create fee entities
+  let poolLpFee = LiquidityPoolFee.load(poolAddress.concat("-lp-fee"));
+  if (!poolLpFee) {
+    poolLpFee = new LiquidityPoolFee(poolAddress.concat("-lp-fee"));
+  }
+
+  let poolProtocolFee = LiquidityPoolFee.load(
     poolAddress.concat("-protocol-fee")
   );
-  const poolTradingFee = new LiquidityPoolFee(
+  if (!poolProtocolFee) {
+    poolProtocolFee = new LiquidityPoolFee(poolAddress.concat("-protocol-fee"));
+  }
+
+  let poolTradingFee = LiquidityPoolFee.load(
     poolAddress.concat("-trading-fee")
   );
+  if (!poolTradingFee) {
+    poolTradingFee = new LiquidityPoolFee(poolAddress.concat("-trading-fee"));
+  }
 
+  // set fee types
   poolLpFee.feeType = LiquidityPoolFeeType.FIXED_LP_FEE;
```

Comment:
Small detail. But how about moving these inside the `if` conditions for new creation of the entity, since these will be unchanging?